glib/src/commonMain/kotlin/io/github/cgr/glib/LazyList.kt

```diff
@@ -0,0 +1,162 @@
+package io.github.cgr.glib
+
+class LazyList(private val generator: Iterator<Any?>) : List<Any?> {
+
+    constructor(generator: () -> Any?) : this(object : Iterator<Any?> {
+        override fun hasNext(): Boolean = true
+
+        override fun next(): Any? = generator()
+    })
+
+    constructor(vararg elements: Any?) : this(elements.iterator())
+
+    constructor(generator: Sequence<Any?>) : this(generator.iterator())
+
+    constructor(generator: Iterable<Any?>) : this(generator.iterator())
+
+    override val size: Int
```

Comment:
Just to be sure I understand this correctly: The program will freeze if we call `.size` on an infinite list, correct? I don't think that's unreasonable behavior, but I do want to make sure I'm reading this correctly.

Reply:
Correct. Thats why I overrode `isEmpty` because it uses `size`, while you can determine if an infinite list is not empty.